bottlecap/src/lifecycle/invocation/triggers/serde_utils.rs

```diff
@@ -0,0 +1,17 @@
+use std::collections::HashMap;
+use std::hash::Hash;
+
+use serde::de::{Deserialize, Deserializer};
+
+/// Deserialize a map, turning `null` into [`HashMap::default`].
+pub(super) fn nullable_map<
+    'de,
+    K: Deserialize<'de> + Eq + Hash,
+    V: Deserialize<'de>,
+    D: Deserializer<'de>,
+>(
+    deserializer: D,
+) -> Result<HashMap<K, V>, D::Error> {
+    let maybe = Option::deserialize(deserializer)?;
+    Ok(maybe.unwrap_or_default())
+}
```

Comment:
Doesn't a `#[serde(default)]` do this?

Not sure tbh, but sounds like a solved problem that should work OOTB

Reply:
It does not -- `#[serde(default)]` makes it valid to NOT have the field at all in the document, and in that case uses the `Default` impl to get a value...

Thing is, some Lambda maps may receive `null` (explicitly), and that is "valid"... I've copied the logic here from the `aws_lambda_events` crate, too.